include/semver/semver.hpp

```diff
@@ -323,13 +335,13 @@ namespace semver
 
             try {
                 if (strict && major_m.matched && minor_m.matched && patch_m.matched) {
-                    major = std::stoul(major_m);
-                    minor = std::stoul(minor_m);
-                    patch = std::stoul(patch_m);
+                    major = (BuildIdentifierPart) parse_build_identifier_part(major_m);
+                    minor = (BuildIdentifierPart) parse_build_identifier_part(minor_m);
+                    patch = (BuildIdentifierPart) parse_build_identifier_part(patch_m);
                 } else if (!strict && major_m.matched) {
-                    major = std::stoul(major_m);
-                    minor = minor_m.matched ? std::stoul(minor_m) : 0;
-                    patch = patch_m.matched ? std::stoul(patch_m) : 0;
+                    major = (BuildIdentifierPart) parse_build_identifier_part(major_m);
+                    minor = minor_m.matched ? (BuildIdentifierPart) parse_build_identifier_part(minor_m) : 0;
+                    patch = patch_m.matched ? (BuildIdentifierPart) parse_build_identifier_part(patch_m) : 0;
```

Comment:
`parse_buld_identifier_part()` already returns `BuildIdentifierPart` so we don't need these type casts.

Reply:
right, good catch... its a leftovers